### PR TITLE
Add i.MX PTA for OCOTP and DIGPROG

### PR DESCRIPTION
**Pull request by clementfaure**

### CODE REVIEW

File: core/pta/imx/ocotp.c [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+						   TEE_PARAM_TYPE_NONE);
+
+	if (param_types != exp_param_types)
+		return TEE_ERROR_BAD_PARAMETERS;
+
+	params[1].value.a = 0;
+	params[1].value.b = 0;
+
+	ret = imx_ocotp_read(params[0].value.a, params[0].value.b, &val);
+	if (ret == TEE_SUCCESS)
```
Comment (jforissier) [inline core/pta/imx/ocotp.c:55]: `if (!ret)`

File: lib/libutee/include/pta_imx_digprog.h [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+#ifndef __PTA_IMX_DIGPROG_H__
+#define __PTA_IMX_DIGPROG_H__
+
+#define PTA_DIGPROG_UUID \
+	{ \
+		0xbfdbe409, 0x3302, 0x4abf, \
+		{ \
+			0x99, 0xbc, 0x5f, 0x62, 0x32, 0x26, 0xa7, 0x39 \
+		} \
+	}
```
Comment (jforissier) [inline lib/libutee/include/pta_imx_digprog.h:14]: Please format like the other `pta_*.h` files, and same for OCOTP

File: core/pta/imx/digprog.c [hunk 3]
```diff
@@ -0,0 +1,32 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 NXP
+ */
+#include "imx.h"
+#include "kernel/pseudo_ta.h"
+#include "pta_imx_digprog.h"
```
Comment (jforissier) [inline core/pta/imx/digprog.c:7]: `<...>` everywhere

**Comment by jforissier** [comment]
With my comments addressed:

`Acked-by: Jerome Forissier <jerome.forissier@linaro.org>`

File: core/pta/imx/ocotp.c [hunk 4]
```diff
@@ -0,0 +1,77 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2021 NXP
+ */
+#include <kernel/pseudo_ta.h>
+#include <kernel/tee_common_otp.h>
+#include <pta_imx_ocotp.h>
+#include <drivers/imx_ocotp.h>
```
[the remark this answers is not in the document]
Comment (clementfaure) [reply core/pta/imx/ocotp.c:8]: Ok

File: core/pta/imx/ocotp.c [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+
+#define OCOTP_PTA_NAME "ocotp.pta"
+
+static TEE_Result chip_uid(uint32_t param_types,
+			   TEE_Param params[TEE_NUM_PARAMS])
+{
+	uint32_t exp_param_types = 0;
+	uint8_t val[IMX_UID_SIZE] = {};
```
[the remark this answers is not in the document]
Comment (clementfaure) [reply core/pta/imx/ocotp.c:16]: Ok

File: core/pta/imx/ocotp.c [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+	exp_param_types = TEE_PARAM_TYPES(TEE_PARAM_TYPE_MEMREF_OUTPUT,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE,
+					  TEE_PARAM_TYPE_NONE);
```
[the remark this answers is not in the document]
Comment (clementfaure) [reply core/pta/imx/ocotp.c:21]: Ok

File: core/pta/imx/ocotp.c (hunk 1 above)
Comment (clementfaure) [reply core/pta/imx/ocotp.c:55]: Ok

File: lib/libutee/include/pta_imx_digprog.h (hunk 2 above)
Comment (clementfaure) [reply lib/libutee/include/pta_imx_digprog.h:14]: Ok

File: core/pta/imx/digprog.c (hunk 3 above)
Comment (clementfaure) [reply core/pta/imx/digprog.c:7]: Ok

**Comment by clementfaure** [comment]
@jforissier Comments addressed and tag applied

**Pull request merged**